packages/bpk-component-select/src/BpkSelect.js

```diff
@@ -15,25 +15,46 @@
  * See the License for the specific language governing permissions and
  * limitations under the License.
  */
+/* @flow */
 
-import PropTypes from 'prop-types';
+import PropTypes, { type Node } from 'prop-types';
 import React from 'react';
 import { cssModules } from 'bpk-react-utils';
 
 import STYLES from './bpk-select.scss';
 
 const getClassName = cssModules(STYLES);
 
-const BpkSelect = props => {
+export type Props = {
+  id: string,
+  name: string,
+  value: string,
+  className: ?string,
+  docked: boolean,
+  dockedFirst: boolean,
+  dockedLast: boolean,
+  dockedMiddle: boolean,
+  image: ?() => Node,
```

Comment:
I'm not convinced that this needs to be a render prop. 

I know that it's so users can change it based on what's selected, but they can do that already by wrapping it in a parent component which stores the selected item in state, which I imagine is how this component is already used by consumers. The fact that we aren't passing any arguments into the render function indicates to me that it isn't necessary.

Having said that, the convention for render props is to name them `renderThing` - so I think `renderImage` would be a better name here **if we do keep it as a render prop**.

![](https://i.imgflip.com/2bkkzg.jpg)

Reply:
![2bkm4v](https://user-images.githubusercontent.com/3579758/40909676-0e517aa6-67e2-11e8-8740-34c0c353c100.jpg)